extensions/windows-event-log/CollectorInitiatedSubscription.cpp

```diff
@@ -0,0 +1,777 @@
+  supportsExpressionLanguage(true)->
+  build());
+
+static SupportedProperty s_maxBufferSize(
+  core::PropertyBuilder::createProperty("Max Buffer Size")->
+  isRequired(true)->
+  withDefaultValue<core::DataSizeValue>("1 MB")->
+  withDescription(
+    "The individual Event Log XMLs are rendered to a buffer."
+    " This specifies the maximum size in bytes that the buffer will be allowed to grow to. (Limiting the maximum size of an individual Event XML.)")->
+  build());
+
+static SupportedProperty s_inactiveDurationToReconnect(
+  core::PropertyBuilder::createProperty("Inactive Duration To Reconnect")->
+  isRequired(true)->
+  withDefaultValue<core::TimePeriodValue>("10 min")->
+  withDescription(
+    "If no new event logs are processed for the specified time period,"
+    " this processor will try reconnecting to recover from a state where any further messages cannot be consumed."
+    " Such situation can happen if Windows Event Log service is restarted, or ERROR_EVT_QUERY_RESULT_STALE (15011) is returned."
+    " Setting no duration, e.g. '0 ms' disables auto-reconnection.")->
+  build());
+
+//! Supported Relationships
+static core::Relationship s_success("success", "Relationship for successfully consumed events.");
+
+
+CollectorInitiatedSubscription::CollectorInitiatedSubscription(const std::string& name, utils::Identifier uuid)
+  : core::Processor(name, uuid), logger_(logging::LoggerFactory<CollectorInitiatedSubscription>::getLogger()) {
+  char buff[MAX_COMPUTERNAME_LENGTH + 1];
+  DWORD size = sizeof(buff);
+  if (GetComputerName(buff, &size)) {
+    computerName_ = buff;
+  } else {
+    logWindowsError(__LINE__, "GetComputerName");
+  }
+}
+
+void CollectorInitiatedSubscription::initialize() {
+  //! Set the supported properties
+  setSupportedProperties(s_supportedProperties);
+
+  //! Set the supported relationships
+  setSupportedRelationships({s_success});
+}
+
+void CollectorInitiatedSubscription::onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory> &sessionFactory) {
+  if (subscriptionHandle_) {
+    logger_->log_error("Processor already subscribed to Event Log, expected cleanup to unsubscribe.");
+  } else {
+    sessionFactory_ = sessionFactory;
+
+    if (!createSubscription(context))
+      return;
+    if (!checkSubscriptionRuntimeStatus())
+      return;
+
+    subscribe(context);
+  }
+}
+
+void CollectorInitiatedSubscription::onTrigger(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSession> &session) {
+  if (!subscriptionHandle_) {
+    if (!subscribe(context)) {
+      context->yield();
+      return;
+    }
+  }
+
+  // Check subscription runtime status.
+  checkSubscriptionRuntimeStatus();
+
+  const auto flowFileCount = processQueue(session);
+
+  const auto now = GetTickCount();
```

Comment:
As far as I remember gettickkount _overflows_ after a while (at about 50 days). Running on Windows servers that might be exceeded causing issues here. 
I don't care about being platform-independent here given this code is only useful on Win, but std::chrono might be a safer choice for timing. 
If you are aware of Windows-specific cons of chrono, please don't hesitate to share!

Reply:
Good point from Arpad. My addition: utils/TimeUtil.h provides getTimeMillis which uses chrono, so that is an option, Alex, though if that if GetTickCount64 is an option I guess.